satpy/tests/reader_tests/test_tropomi_l2.py

```diff
@@ -0,0 +1,174 @@
+            file_content['PRODUCT/longitude'] = DEFAULT_FILE_DATA
+
+            if 'NO2' in filename:
+                file_content['PRODUCT/nitrogen_dioxide_total_column'] = DEFAULT_FILE_DATA
+            if 'SO2' in filename:
+                file_content['PRODUCT/sulfurdioxide_total_vertical_column'] = DEFAULT_FILE_DATA
+
+            for k in list(file_content.keys()):
+                if not k.startswith('PRODUCT'):
+                    continue
+                file_content[k + '/shape'] = DEFAULT_FILE_SHAPE
+
+            # convert to xarrays
+            for key, val in file_content.items():
+                if isinstance(val, np.ndarray):
+                    if val.ndim > 1:
+                        file_content[key] = DataArray(val, dims=('y', 'x'))
+                    else:
+                        file_content[key] = DataArray(val)
+
+        else:
+            print('filetype_info is: ', filetype_info)
+            print('filetype is: ', filetype_info['file_type'])
+            assert False
+
+        return file_content
+
+
+class TestTROPOMIL2Reader(unittest.TestCase):
+    """Test TROPOMI L2 Reader"""
+    yaml_file = "tropomi_l2.yaml"
+
+    def setUp(self):
+        """Wrap NetCDF4 file handler with our own fake handler"""
+        from satpy.config import config_search_paths
+        from satpy.readers.tropomi_l2 import TROPOMIL2FileHandler
+        self.reader_configs = config_search_paths(os.path.join('readers', self.yaml_file))
+        # http://stackoverflow.com/questions/12219967/how-to-mock-a-base-class-with-python-mock-library
+        self.p = mock.patch.object(TROPOMIL2FileHandler, '__bases__', (FakeNetCDF4FileHandlerTL2,))
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def tearDown(self):
+        """Stop wrapping the NetCDF4 file handler"""
+        self.p.stop()
+
+    def test_init(self):
+        """Test basic initialization of this reader."""
+        from satpy.readers import load_reader
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'S5P_OFFL_L2__NO2____20180709T170334_20180709T184504_03821_01_010002_20180715T184729.nc',
+        ])
+        self.assertTrue(len(loadables), 1)
+        r.create_filehandlers(loadables)
+        # make sure we have some files
+        self.assertTrue(r.file_handlers)
+
+    def test_load_available_datasets(self, configured_datasets=None):
+        """Test loading datasets."""
+        from satpy.readers import load_reader
+        from satpy import DatasetID
+        print("test_load_available_datasets begin...")
+        r = load_reader(self.reader_configs)
+        loadables = r.select_files_from_pathnames([
+            'S5P_OFFL_L2__NO2____20180709T170334_20180709T184504_03821_01_010002_20180715T184729.nc',
+            ])
+        r.create_filehandlers(loadables)
+
+        # Determine shape of the geolocation data (lat/lon)
```

Comment:
Everything below this looks like a copy/paste from your `available_datasets`? If not, what is it doing? Can't you test loading datasets by looking at `available_dataset_names` or something similar?